client/my-sites/site-settings/form-general.jsx

```diff
@@ -421,6 +429,27 @@ module.exports = React.createClass( {
 		);
 	},
 
+	renderDomainNudge() {
+		// do not nudge if the site already has a custom domain
+		if ( siteHasCustomDomain( this.props.site ) ) {
+			return;
+		}
+
+		// render nudge if the site plan is not free
+		if ( ! siteHasFreePlan( this.props.site.plan ) ) {
+			return;
+		}
+
+		return (
+			<UpgradeNudge
+				title={ this.translate( 'Add A Custom Domain' ) }
+				message={ this.translate( 'Upgrade now and get a free custom domain.' ) }
+				href={ `/plans/compare/${ CUSTOM_DOMAIN }/${ this.props.site.domain }` }
```

Comment:
You don't need to construct the plans compare url, if you set a `feature` prop that would work by default.